src/common/function/src/scalars/geo/h3.rs

```diff
@@ -152,20 +164,33 @@ impl Function for H3LatLngToCellString {
         Signature::one_of(signatures, Volatility::Stable)
     }
 
-    fn eval(&self, _func_ctx: &FunctionContext, columns: &[VectorRef]) -> Result<VectorRef> {
-        ensure_columns_n!(columns, 3);
-
-        let lat_vec = &columns[0];
-        let lon_vec = &columns[1];
-        let resolution_vec = &columns[2];
+    fn invoke_with_args(
+        &self,
+        args: ScalarFunctionArgs,
+    ) -> datafusion_common::Result<ColumnarValue> {
+        let args = ColumnarValue::values_to_arrays(&args.args)?;
+        let [lat_vec, lon_vec, resolution_vec] = utils::take_function_args(self.name(), args)?;
 
         let size = lat_vec.len();
-        let mut results = StringVectorBuilder::with_capacity(size);
+        let mut builder = StringViewBuilder::with_capacity(size);
 
         for i in 0..size {
-            let lat = lat_vec.get(i).as_f64_lossy();
-            let lon = lon_vec.get(i).as_f64_lossy();
-            let r = value_to_resolution(resolution_vec.get(i))?;
+            let lat = ScalarValue::try_from_array(&lat_vec, i).map(|v| {
+                if let ScalarValue::Float64(x) = v {
+                    x
+                } else {
+                    None
+                }
+            })?;
```

Comment:
Getting a ScalarValue each time may be inefficient. Should we optimize it in this PR or do it later?